smart/model_assembly.py

```diff
@@ -654,6 +678,56 @@ def from_file(cls, name, sampling_file, unit, group="", notes="", use_preintegra
 
         return parameter
 
+    @classmethod
+    def from_xdmf(
+        cls, name, xdmf_file, unit, compartment, group="", notes="", use_preintegration=False
+    ):
+        """ "
+        Data read in from an xdmf/h5 file pairing
+        """
+        if isinstance(xdmf_file, str):
+            xdmf_file = Path(xdmf_file)
+        assert xdmf_file.is_file(), f"{str(xdmf_file)} could not be found to load parameter"
+
+        logger.debug(f"Loading initial condition for {name} from file")
+        if xdmf_file.suffix == ".xdmf":
+            xdmfCur = str(xdmf_file)
+            h5Cur = xdmfCur[0:-4] + "h5"
+        else:
+            raise TypeError(f"{str(xdmf_file)} cannot be used for loading parameter, must be xdmf")
+
+        # load in xdmf file
+        logger.info(f"Loading in data for parameter {name}", extra=dict(format_type="log"))
+
+        if use_preintegration:
+            logger.warning(
+                f"Setting use_preintegration to False for parameter {name}."
+                "Not currently implemented for parameters loaded from xdmf"
+            )
+            use_preintegration = False
+        parameter = cls(
+            name,
+            0.0,
+            unit,
+            group=group,
+            notes=notes,
+            use_preintegration=use_preintegration,
+        )
+        parameter.compartment = compartment
+        # initialize instance
+        parameter.xdmf_file = xdmfCur
+        parameter.h5_file = h5Cur
+        parameter.is_time_dependent = True
+        parameter.is_space_dependent = True
+        parameter.type = ParameterType.from_xdmf
```

Comment:
Is it possible to add these arguments to the constructor instead, so that we know that all Parameter objets have these attributes?